### PR TITLE
feat: partial wasm32-wasi compilation

### PR DESCRIPTION
**Pull request by jtguibas**
The goal of this PR is to set a foundation for supporting wasm32-wasi compilation of reth crates. 

The currently supported crates are: reth-codecs, reth-consensus, reth-db, reth-db-api, reth-ethereum-forks, reth-evm, reth-evm-ethereum, reth-network-peers, reth-primitives, reth-primitives-traits, and reth-revm with no default features but std. Overtime, we can add support for more crates as needed.

Merging this PR succesfully requires having updated releases of `revm` and `revm-inspector` following the merge of this PR: https://github.com/bluealloy/revm/pull/1601 due to the usage of `kzg-rs` instead of `c-kzg` to get compilation to work successfully.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -363,27 +363,26 @@ reth-stages-api = { path = "crates/stages/api" }
...
 reth-tokio-util = { path = "crates/tokio-util" }
 reth-tracing = { path = "crates/tracing" }
 reth-transaction-pool = { path = "crates/transaction-pool" }
-reth-trie = { path = "crates/trie/trie" }
+reth-trie = { path = "crates/trie/trie", default-features = false }
 reth-trie-common = { path = "crates/trie/common" }
 reth-trie-parallel = { path = "crates/trie/parallel" }
 
 # revm
-revm = { version = "11.0.0", features = [
```
Comment (jtguibas) [inline Cargo.toml:378]: TODO: Update once https://github.com/bluealloy/revm/pull/1601 merged into `revm` and `revm-inspectors` is also updated.

File: Cargo.toml
Comment (mattsse) [inline Cargo.toml:1]: could you submit this separately?

File: crates/storage/db/src/utils.rs [hunk 2]
```diff
@@ -16,6 +17,11 @@ pub(crate) fn default_page_size() -> usize {
     os_page_size.clamp(min_page_size, libmdbx_max_page_size)
 }
 
+#[cfg(not(feature = "page_size"))]
+pub(crate) fn default_page_size() -> usize {
+    0
```
Comment (mattsse) [inline crates/storage/db/src/utils.rs:22]: this is a bit dangerous

this is used as an option, so I think we want to make the function that is used to configure the page size return an option instead

https://github.com/paradigmxyz/reth/blob/fc4c037e60b623b81b296fe9242fa905ff36b89a/crates/storage/libmdbx-rs/src/environment.rs#L632-L637

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive, but I'd like to do this in multiple steps

**Comment by shekhirin** [comment]
Superseded by https://github.com/paradigmxyz/reth/issues/10088

**Pull request closed without merging**